File: apps/web/api/health.ts
import type { VercelRequest, VercelResponse } from '@vercel/node'

export default async function handler(req: VercelRequest, res: VercelResponse) {
  // Enable CORS
  res.setHeader('Access-Control-Allow-Credentials', 'true')
  res.setHeader('Access-Control-Allow-Origin', '*')
  res.setHeader('Access-Control-Allow-Methods', 'GET,OPTIONS')
  
  if (req.method === 'OPTIONS') {
    res.status(200).end()
    return
  }

  return res.status(200).json({
    status: 'healthy',
    service: 'AGENTLAND.SAARLAND Frontend API',
    timestamp: new Date().toISOString(),
    version: '1.0.0'
  })
}

File: apps/api/api/health.py
"""
Health Check API for Vercel deployment
"""
from http.server import BaseHTTPRequestHandler
import json
import os

class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        self.send_response(200)
        self.send_header('Content-type', 'application/json')
        self.send_header('Access-Control-Allow-Origin', '*')
        self.end_headers()
        
        response = {
            "status": "healthy",
            "service": "AGENTLAND.SAARLAND API",
            "region": "Saarland",
            "deployment": "vercel",
            "deepseek_configured": bool(os.getenv("DEEPSEEK_API_KEY")),
            "vectordb_configured": bool(os.getenv("PINECONE_API_KEY")),
        }
        
        self.wfile.write(json.dumps(response).encode())
        return